aspnetcore/mvc/controllers/actions.md

```diff
@@ -14,80 +14,86 @@ uid: mvc/controllers/actions
 
 ## Defining Actions
 
-Any public method on a controller type is an action. Parameters on actions are bound to request data and validated using [model binding](../models/model-binding.md).
+Any public method on a controller is an action. Parameters on actions are bound to request data and are validated using [model binding](xref:mvc/models/model-binding). Model validation can be accomplished by explicitly verifying the `ModelState.IsValid` property is true.
 
->[!WARNING]
-> Action methods that accept parameters should verify the `ModelState.IsValid` property is true.
+Action methods should contain logic for mapping a request to a business concern. Business concerns should typically be represented as services that the controller accesses through [dependency injection](xref:mvc/controllers/dependency-injection). Actions then map the result of the business action to an application state.
 
-Action methods should contain logic for mapping an incoming request to a business concern. Business concerns should typically be represented as services that your controller accesses through [dependency injection](dependency-injection.md). Actions then map the result of the business action to an application state.
-Actions can return anything, but frequently will return an instance of `IActionResult` (or `Task<IActionResult>` for async methods) that produces a response. The action method is responsible for choosing *what kind of response*; the action result *does the responding*.
+Actions can return anything, but frequently return an instance of `IActionResult` (or `Task<IActionResult>` for async methods) that produces a response. The action method is responsible for choosing *what kind of response*. The action result *does the responding*.
 
 ### Controller Helper Methods
 
-Although not required, most developers will want to have their controllers inherit from the base `Controller` class. Doing so provides controllers with access to many properties and helpful methods, including the following helper methods designed to assist in returning various responses:
+Controllers usually inherit from [Controller](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.controller), although this is not required. Deriving from `Controller` provides access to three categories of helper methods designed to return specific response types:
 
-**[View](../views/index.md)**
+#### 1. An empty response body
 
-Returns a view that uses a model to render HTML. Example: `return View(customer);`
+No `Content-Type` HTTP response header is included, since the response body lacks content to describe.
 
-**HTTP Status Code**
+There are two response types within this category: Redirect and HTTP Status Code. The Redirect response type differs from the HTTP Status Code type primarily in the addition of a `Location` HTTP response header.
```

Comment:
Suggest reversing the bullets below and moving the second sentence under the **Redirect** bullet